### PR TITLE
KTLN-380: Added unit test for MaxValue package

### PR DESCRIPTION
**Pull request by diegotorrespy**
KTLN-380: Added unit test for MaxValue package

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-collections-4/pom.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <artifactId>core-kotlin-modules</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+
+    <dependencies>
+    </dependencies>
+
+    <properties>
+
+    </properties>
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/pom.xml:21]: Remove the unused/empty sections

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.jupiter.api.Test
+
+class MaxValueArrayTest {
+
+    @Test
+    fun givenNonEmptyArray_whenGetMaxOrNull_thenMaxValueIsReturned() {
+
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:12]: In this and the other tests, remove the empty line at the beginning. One of the guidelines prescribes not to have empty lines after an opening curly brace or before a closing one.

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @Test
+    fun givenNonEmptyArray_whenGetMaxBy_thenMaxValueIsReturned() {
+
+        val array = listOf(
+            Person("Alan", 34, Height(5, 10)),
+            Person("Jason", 45, Height(5, 11)),
+            Person("Joe", 28, Height(5, 9))
+        )
+        val max = array.maxByOrNull { p -> p.age }
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:27]: Insert a blank line here (put blank line between the given/when and when/then part of a test)

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Test
+    fun givenNonEmptyArray_whenGetMaxWith_thenMaxValueIsReturned() {
+
+        val array = listOf(
+            Person("Alan", 34, Height(5, 10)),
+            Person("Jason", 45, Height(5, 13)),
+            Person("Joe", 45, Height(5, 11))
+        )
+        val max = array.maxWith(compareBy({ it.height.feet }, { it.height.inches }))
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:44]: insert blank line

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Test
+    fun givenNonEmptyArrayWithMultipleMaxCandidate_whenGetMaxWith_thenMaxValueIsReturned() {
+
+        val array = listOf(
+            Person("Alan", 34, Height(5, 10)),
+            Person("Jason", 45, Height(5, 11)),
+            Person("Joe", 45, Height(5, 11))
+        )
+        val max = array.maxWith(compareBy({ it.height.feet }, { it.height.inches }))
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:58]: insert blank line

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Test
+    fun givenNonEmptyArray_whenGetBySorting_thenMaxValueShouldBeReturned() {
+
+        val array = listOf(
+            Person("Alan", 34, Height(5, 10)),
+            Person("Jason", 45, Height(5, 11)),
+            Person("Joe", 28, Height(5, 9))
+        )
+        val max = array.sortedByDescending { p -> p.age }.first()
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:72]: insert blank line

File: core-kotlin-modules/pom.xml [hunk 7]
```diff
@@ -28,6 +28,7 @@
         <module>core-kotlin-collections</module>
         <module>core-kotlin-collections-2</module>
         <module>core-kotlin-collections-3</module>
+				<module>core-kotlin-collections-4</module>
```
Comment (KevinGilmore) [inline core-kotlin-modules/pom.xml:31]: use spaces to indent, not tabs

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+        val array = listOf(
+            Person("Alan", 34, Height(5, 10)),
+            Person("Jason", 45, Height(5, 11)),
+            Person("Joe", 28, Height(5, 9))
+        )
+        val max = array.maxByOrNull { p -> p.age }
+
+        assertThat(max).isNotNull()
+        if (max != null) {
```
Comment (KevinGilmore) [inline core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt:30]: I'm not much of a Kotlin guy, but are these null checks necessary?

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
See comments regarding blank lines and null check

File: core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt (hunk 8 above)
Comment (diegotorrespy) [reply core-kotlin-modules/core-kotlin-collections-4/src/test/kotlin/com/baeldung/maxvalue/MaxValueArrayTest.kt]: Yeah, it was redundant. Thanks

**Pull request merged**